Fuyu.Common/Networking/WsContext.cs

```diff
@@ -0,0 +1,79 @@
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Net;
+using System.Net.WebSockets;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+using Fuyu.Common.Compression;
+using Fuyu.Common.Serialization;
+
+namespace Fuyu.Common.Networking
+{
+    public class WsContext : Context
+    {
+        private const int _bufferSize = 32000;
+        private readonly WebSocket _ws;
+
+        public Func<WsContext, Task> OnCloseAsync;
+        public Func<WsContext, string, Task> OnTextAsync;
+        public Func<WsContext, byte[], Task> OnBinaryAsync;
+
+        public WsContext(HttpListenerRequest request, HttpListenerResponse response, WebSocket ws) : base(request, response)
+        {
+            _ws = ws;
+        }
+
+        public bool IsOpen()
+        {
+            return _ws.State == WebSocketState.Open;
+        }
+
+        // TODO:
+        // * use System.Buffers.ArrayPool for receiveBuffer
+        // -- seionmoya, 2024/09/09
+        public async Task ReceiveAsync()
+        {
+            var receiveBuffer = new byte[_bufferSize];
+            var receiveResult = await _ws.ReceiveAsync(new ArraySegment<byte>(receiveBuffer), CancellationToken.None);
+
+            switch (receiveResult.MessageType)
+            {
+                case WebSocketMessageType.Close:
+                    await OnCloseAsync(this);
+                    await CloseAsync();
+                    break;
+
+                case WebSocketMessageType.Text:
+                    var text = Encoding.UTF8.GetString(receiveBuffer);
+                    await OnTextAsync(this, text);
+                    break;
+
+                case WebSocketMessageType.Binary:
+                    var data = new byte[receiveResult.Count];
```

Comment:
Move lines 54 and 55 to line 40. Right now on line 49 OnTextAsync is receiving the entire 32kb buffer.